# Cleansing Data Youtube Rewind 2020

#### Import Library

In [5]:
from dotenv import load_dotenv
from pymongo import MongoClient
import os
import pandas as pd

#### Import url dari .env

In [6]:
# Memuat value dari file .env
load_dotenv()

mongodb_url = os.getenv('URL_SANDY')
local_url = os.getenv('URL_LOCAL')

In [7]:
database_name = "youtube_rewind_indonesia"
collection_from = "collect_2020"
collection_name = "cleansing_2020"

#### Import dataset dari MongoDB

In [8]:
# Koneksi ke MongoDB
client = MongoClient(local_url)

# Pilih database dan koleksi
db = client[database_name]
collection = db[collection_from]

# Query data dari koleksi
cursor = collection.find()

# Konversi cursor ke list dan kemudian ke DataFrame
data = list(cursor)
df = pd.DataFrame(data)

# Menutup koneksi
client.close()

In [9]:
df

_id                 channelId      videoId  \
0       66516226b0b542363152a0f6  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
1       66516226b0b542363152a0f7  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
2       66516226b0b542363152a0f8  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
3       66516226b0b542363152a0f9  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
4       66516226b0b542363152a0fa  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
...                          ...                       ...          ...   
333495  665165f5b0b542363157b7ad  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
333496  665165f5b0b542363157b7ae  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
333497  665165f5b0b542363157b7af  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
333498  665165f5b0b542363157b7b0  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
333499  665165f5b0b542363157b7b1  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   

                                              textDisplay  \
0       We, from the Rewind Indonesia 2020 team respec...   
1                                                 Afa ada   
2                                                   ❤❤❤❤❤   
3                                          Gw kangen 2020   
4        Kerja bagus ini sunggnh bagus woh semoga di baca   
...                                                   ...   
333495  <a href="https://www.youtube.com/watch?v=LhYfs...   
333496                             WOII KEREN BANGEETTTTT   
333497                             Masih best rewind 2018   
333498  The epic moments <a href="http://www.youtube.c...   
333499                      merinding banget sumpah 😔❤❤❤❤   

                                             textOriginal  \
0       We, from the Rewind Indonesia 2020 team respec...   
1                                                 Afa ada   
2                                                   ❤❤❤❤❤   
3                                          Gw kangen 2020   
4        Kerja bagus ini sunggnh bagus woh semoga di baca   
...                                                   ...   
333495                                    6:40 merinding😶   
333496                             WOII KEREN BANGEETTTTT   
333497                             Masih best rewind 2018   
333498                           The epic moments #RI2020   
333499                      merinding banget sumpah 😔❤❤❤❤   

               authorDisplayName  \
0             @IndonesianCreator   
1          @muhammadfahryzal3498   
2              @juliaazzahra7753   
3               @user-bw3ze4hc8s   
4               @user-ex8yk7tp1b   
...                          ...   
333495      @ismanoerfatah._1565   
333496  @rahmafitriyanixotkp3148   
333497       @ferzaexclusive6434   
333498             @sulfadli2171   
333499         @nadirasheril1886   

                                    authorProfileImageUrl  \
0       https://yt3.ggpht.com/fO77gm9qrRYfPRp2NDPodQ9i...   
1       https://yt3.ggpht.com/ytc/AIdro_msn1kSAZwJIxjE...   
2       https://yt3.ggpht.com/ytc/AIdro_kQOzuPqs3TbTAB...   
3       https://yt3.ggpht.com/ytc/AIdro_nxHXmfTkoLg7t9...   
4       https://yt3.ggpht.com/ytc/AIdro_nhfd1E23j6qheN...   
...                                                   ...   
333495  https://yt3.ggpht.com/ytc/AIdro_kGjodx108CD4Ay...   
333496  https://yt3.ggpht.com/ytc/AIdro_kVt-2KnstlTCEK...   
333497  https://yt3.ggpht.com/ytc/AIdro_nLVuCODvdNq-18...   
333498  https://yt3.ggpht.com/hL8b3eYnmNVF3zxdio4XCk9D...   
333499  https://yt3.ggpht.com/ytc/AIdro_kBJ6lXgh-zF7ee...   

                                       authorChannelUrl  \
0             http://www.youtube.com/@IndonesianCreator   
1          http://www.youtube.com/@muhammadfahryzal3498   
2              http://www.youtube.com/@juliaazzahra7753   
3               http://www.youtube.com/@user-bw3ze4hc8s   
4               http://www.youtube.com/@user-ex8yk7tp1b   
...                                                 ...   
333495      http://www.youtube.com/@ismanoerfatah._1565   
333496  http://www.youtu

#### Menghapus data text yang duplikat

In [10]:
df = df.drop_duplicates(subset=['textOriginal'])

df.shape

(277039, 14)

#### Menghapus data author yang duplikat

In [11]:
df = df.drop_duplicates(subset=['authorChannelUrl'])

df.shape

(244034, 14)

#### Menghilangkan atribut yang tidak diperlukan

Karena hanya menganalisis komentar, maka atribut yang disisakan hanya _id dan textOriginal

In [12]:
# df = df[['_id', 'textOriginal']]
df

_id                 channelId      videoId  \
0       66516226b0b542363152a0f6  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
1       66516226b0b542363152a0f7  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
2       66516226b0b542363152a0f8  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
3       66516226b0b542363152a0f9  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
4       66516226b0b542363152a0fa  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
...                          ...                       ...          ...   
333495  665165f5b0b542363157b7ad  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
333496  665165f5b0b542363157b7ae  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
333497  665165f5b0b542363157b7af  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
333498  665165f5b0b542363157b7b0  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   
333499  665165f5b0b542363157b7b1  UC-z1ubmS-iHMVr83EEwRS2g  LhYfsKc0_tA   

                                              textDisplay  \
0       We, from the Rewind Indonesia 2020 team respec...   
1                                                 Afa ada   
2                                                   ❤❤❤❤❤   
3                                          Gw kangen 2020   
4        Kerja bagus ini sunggnh bagus woh semoga di baca   
...                                                   ...   
333495  <a href="https://www.youtube.com/watch?v=LhYfs...   
333496                             WOII KEREN BANGEETTTTT   
333497                             Masih best rewind 2018   
333498  The epic moments <a href="http://www.youtube.c...   
333499                      merinding banget sumpah 😔❤❤❤❤   

                                             textOriginal  \
0       We, from the Rewind Indonesia 2020 team respec...   
1                                                 Afa ada   
2                                                   ❤❤❤❤❤   
3                                          Gw kangen 2020   
4        Kerja bagus ini sunggnh bagus woh semoga di baca   
...                                                   ...   
333495                                    6:40 merinding😶   
333496                             WOII KEREN BANGEETTTTT   
333497                             Masih best rewind 2018   
333498                           The epic moments #RI2020   
333499                      merinding banget sumpah 😔❤❤❤❤   

               authorDisplayName  \
0             @IndonesianCreator   
1          @muhammadfahryzal3498   
2              @juliaazzahra7753   
3               @user-bw3ze4hc8s   
4               @user-ex8yk7tp1b   
...                          ...   
333495      @ismanoerfatah._1565   
333496  @rahmafitriyanixotkp3148   
333497       @ferzaexclusive6434   
333498             @sulfadli2171   
333499         @nadirasheril1886   

                                    authorProfileImageUrl  \
0       https://yt3.ggpht.com/fO77gm9qrRYfPRp2NDPodQ9i...   
1       https://yt3.ggpht.com/ytc/AIdro_msn1kSAZwJIxjE...   
2       https://yt3.ggpht.com/ytc/AIdro_kQOzuPqs3TbTAB...   
3       https://yt3.ggpht.com/ytc/AIdro_nxHXmfTkoLg7t9...   
4       https://yt3.ggpht.com/ytc/AIdro_nhfd1E23j6qheN...   
...                                                   ...   
333495  https://yt3.ggpht.com/ytc/AIdro_kGjodx108CD4Ay...   
333496  https://yt3.ggpht.com/ytc/AIdro_kVt-2KnstlTCEK...   
333497  https://yt3.ggpht.com/ytc/AIdro_nLVuCODvdNq-18...   
333498  https://yt3.ggpht.com/hL8b3eYnmNVF3zxdio4XCk9D...   
333499  https://yt3.ggpht.com/ytc/AIdro_kBJ6lXgh-zF7ee...   

                                       authorChannelUrl  \
0             http://www.youtube.com/@IndonesianCreator   
1          http://www.youtube.com/@muhammadfahryzal3498   
2              http://www.youtube.com/@juliaazzahra7753   
3               http://www.youtube.com/@user-bw3ze4hc8s   
4               http://www.youtube.com/@user-ex8yk7tp1b   
...                                                 ...   
333495      http://www.youtube.com/@ismanoerfatah._1565   
333496  http://www.youtu

#### Menghapus nilai yang kosong

In [13]:
df = df.dropna()

# Melihat data yang kosong
df.isnull().sum()

_id                      0
channelId                0
videoId                  0
textDisplay              0
textOriginal             0
authorDisplayName        0
authorProfileImageUrl    0
authorChannelUrl         0
authorChannelId          0
canRate                  0
viewerRating             0
likeCount                0
publishedAt              0
updatedAt                0
dtype: int64

Dari yang awalnya berjumlah 333500 row berkurang menjadi 244034 row

#### Mengimpan Clean Comments ke MongoDB Local

In [14]:
# Koneksi ke MongoDB
client = MongoClient(local_url)
db = client[database_name]
collection = db[collection_name]

# Konversi DataFrame ke bentuk dictionary
records = df.to_dict(orient='records')

# Menyisipkan data ke MongoDB
collection.insert_many(records)

print("Data berhasil diimpor ke MongoDB Local.")

Data berhasil diimpor ke MongoDB Local.
